web-common/src/features/dashboards/pivot/PivotTable.svelte

```diff
@@ -124,26 +137,83 @@
       }
     }
   };
+
+  function calculateFirstColumnWidth(firstColumnName: string) {
+    const rows = $pivotDataStore.data;
+
+    const isTimeDimension = firstColumnName.startsWith("__time");
+
+    // Dates are displayed as shorter values
+    if (isTimeDimension) return MIN_COL_WIDTH;
+
+    const samples = extractSamples(
+      rows.map((row) => row[firstColumnName]),
+    ).filter((v): v is string => typeof v === "string");
+
+    const averageValueLength =
+      samples.reduce((sum, value) => sum + value.length, 0) / rows.length;
+
+    const finalBasis = Math.max(firstColumnName.length, averageValueLength);
+    const pixelLength = finalBasis * 7;
+    const final = Math.max(
+      MIN_COL_WIDTH,
+      Math.min(MAX_INIT_COL_WIDTH, pixelLength + 30),
+    );
+
+    return final;
+  }
+
+  function extractSamples<T>(arr: T[], sampleSize: number = 30) {
+    if (arr.length <= sampleSize) {
+      return arr.slice();
+    }
+
+    const sectionSize = Math.floor(sampleSize / 3);
+
+    const lastSectionSize = sampleSize - sectionSize * 2;
+
+    const first = arr.slice(0, sectionSize);
+
+    let middleStartIndex = Math.floor((arr.length - sectionSize) / 2);
+    const middle = arr.slice(middleStartIndex, middleStartIndex + sectionSize);
+
+    const last = arr.slice(-lastSectionSize);
+
+    return [...first, ...middle, ...last];
+  }
 </script>
 
 <div
   style:--row-height="{ROW_HEIGHT}px"
   style:--header-length="{totalHeaderHeight}px"
+  style:--first-column-width="{firstColumnWidth}px"
   class="overflow-scroll h-fit max-h-full border rounded-md bg-white"
   bind:this={containerRefElement}
   on:scroll={() => handleScroll(containerRefElement)}
 >
   <div style:height="{totalRowSize + totalHeaderHeight}px">
-    <table>
+    <table class:table-fixed={rows.length > 1}>
       <thead>
         {#each headerGroups as headerGroup}
           <tr>
-            {#each headerGroup.headers as header}
+            {#each headerGroup.headers as header, i}
               {@const sortDirection = header.column.getIsSorted()}
               <th
+                class:first-column={hasDimension && i === 0}
                 colSpan={header.colSpan}
-                class:with-row-dimension={rows.length > 1}
+                class:with-row-dimension={hasDimension}
+                class="relative"
               >
+                {#if i === 0 && hasDimension && firstColumnWidth !== undefined}
+                  <Resizer
```

Comment:
Tanstack table has an inbuilt resizing prop -
https://tanstack.com/table/v8/docs/guide/column-sizing

Do you think it makes sense to use that for all the columns?

Reply:
May revisit, but saw frame drops when using the Tanstack table approach.